### PR TITLE
Backend missing

### PR DESCRIPTION
**Pull request by baude**
Disconnect backends
    
    Ideally, the atomic CLI should be able to operate independently
    of the backends it supports.  For example, if dockerd is inactive,
    the ostree backend and atomic cli should still work.
    
    This requires some tweaking to the backendutils code and the work
    flow.  We also need to specifically know if the user passes
    --storage so that we treat that as an explicit override.  The work
    flow is now roughly:
    
    * a default storage can be defined in atomic.conf (was always this way)
    * if not defined, defaults to docker.
    * if --storage is passed, treat explictly and fail if cannot execute
    * if no --storage is specified, use default.  if default is not available, move
     onto the next backend

### CODE REVIEW

**Comment by baude** [comment]
@rhatdan @giuseppe @yuqi-zhang  PTAL

File: Atomic/run.py [hunk 1]
```diff
@@ -90,7 +90,7 @@ def run(self):
                 img_object = db.has_image(self.image)
             except RegistryInspectError:
                 util.write_err("Unable to find image {}".format(self.image))
-
+                return 1
```
Comment (giuseppe) [inline Atomic/run.py]: what does the 1 here mean?

Comment (baude) [inline Atomic/run.py]: mistake, should be a ValueError

**Comment by yuqi-zhang** [comment]
`atomic images prune` is still using the old method of going through all backends
`atomic images info` on an ostree image returns: This Atomic function requires an active docker daemon. Even if "--storage ostree" is specified.

**Comment by baude** [comment]
@yuqi-zhang Yes, that is going to be the case for a little while because we have to use dockerd to obtain the list of registries to search through when a fully-qualified image name is NOT provided.  I have some ideas about trying to incorporate this into skopeo.

**Comment by yuqi-zhang** [comment]
Ok, other than `info` and `prune` everything else looks good. I will close https://github.com/projectatomic/atomic/pull/802 when we merge this.

**Comment by baude** [comment]
@rh-atomic-bot r+

**Comment by baude** [comment]
@rh-atomic-bot r+ 96cc5fb

**Comment by baude** [comment]
@rh-atomic-bot retry

**Pull request merged**